src/MagicImports/Imports.php

```diff
@@ -0,0 +1,249 @@
+     * @return array $models
+     */
+    public function setRelationships(array $models) : array
+    {
+        /**
+         * Get the import structure
+         */
+        $structures = $this->structure->getStructure();
+        $returns = [];
+        foreach ($structures as $modelName => $data) {
+            if(!isset($data['relationships'])){
+                continue;
+            }
+
+            if(!isset($models[$modelName])){
+                continue;
+            }
+            
+            /**
+             * @var all-models $models
+             * @var PrincipalModel $models[$modelName]
+             * @var relationships $data['relationships']
+             */
+            $saveRelationships = $this->saveRelationships($models, $models[$modelName], $data['relationships']);
+            $returns = array_merge($returns, $saveRelationships);
+        }
+
+        return $returns;
+    }
+
+    /**
+     * Save relationships
+     * @param array $models
+     * @param Model $model
+     * @param array $relationships
+     * @return array $returns
+     */
+    public function saveRelationships($models, Model $model, $relationships) : array
+    {
+        $returns = [];
+        foreach ($models as $class => $obj) {
+            $className = get_class($obj);
+           
+            if(!isset($relationships[$className])){
+                continue;
+            }
+
+            $primaryKey = $relationships[$className]['primaryKey'];
+            $relationshipsKey = $relationships[$className]['relationshipsKey'];
+
+            switch ($relationships[$className]['getType']) {
+                case 0:
+                case 1:
+                    $model->{$primaryKey} = $obj->{$relationshipsKey};
+                    $model->save();
+                    $returns[$this->getClassName($model)] = $model;
+                break;
+                case 2:
+                    $obj->{$relationshipsKey} = $model->{$primaryKey};
+                    $obj->save();
+                    $returns[$this->getClassName($obj)] = $obj;
+                break;
+                case 3:
+                    $intermediateModel = $relationships[$className]['intermediateModel'];
+                    $intermediateFields = $relationships[$className]['intermediateFields'];
+                    $intermediateReferencedField = $relationships[$className]['intermediateReferencedField'];
+                    $newObj = new $intermediateModel;
+                    $newObj->{$intermediateFields} = $model->{$primaryKey};
+                    $newObj->{$intermediateReferencedField} = $obj->{$relationshipsKey};
+                    $newObj->save();
+                    $returns[$this->getClassName($newObj)] = $newObj;
+                break;
+                default:
+                    throw new Exception("Error Processing Request", 1);
+                break;
+            }
+        }
+
+        return $returns;
+    }
+
+    public function getClassName($obj)
```

Comment:
@dasdo doc and type , also what is this for? why not use the one I created in baka ?
https://github.com/bakaphp/baka/blob/0.5/src/functions.php#L121